File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hash_map={}

        for index,val in enumerate(nums):
            if (complement:=target-val) in hash_map:
                return [hash_map[complement],index]
            hash_map[val]=index
        return []
       


        

      
        

File: 0001-two-sum/0001-two-sum.java
class Solution{
    public int [] twoSum(int [] nums, int target){
        Map<Integer,Integer> lookUp=new HashMap<>();

        for (int index=0; index<nums.length;index++){
            int compliment=target-nums[index];

            if(lookUp.containsKey(compliment)){
                return new int[] {index, lookUp.get(compliment)};
            }
            lookUp.put(nums[index], index);
        }
        return new int [] {};
    }
}